presto-spi/src/main/java/io/prestosql/spi/block/Block.java

```diff
@@ -169,6 +169,14 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
      */
     long getSizeInBytes();
 
+    /**
+     * Returns the current size in bytes of this block, ignoring any unloaded parts.
+     */
+    default long getLoadedSizeInBytes()
```

Comment:
How do you intend to use this method?
Isn't `getLoadedRetainedSizeInBytes` more important for memory accounting? 

Take a look at `ProjectSelectedPositions#updateRetainedSize` and `PageUtils#recordMaterializedBytes`. Currently we don't account for memory of blocks which are not loaded (e.g. dictionary with not loaded "values" is also not loaded).
We should make the semantics clear. Who is the owner of the block that is partially loaded? Is it the producer of such block or the consumer (for the loaded parts). If the latter, then we would need to take that into account in code (`ProjectSelectedPositions#updateRetainedSize` and `PageUtils#recordMaterializedBytes`).

Reply:
Theoretically it might be easier if producer is owner of partially loaded blocks because it can react to block loads and adjust memory usage accordingly.

For example if consumer is owner of partially loaded blocks and two blocks use the same underlying lazy block (e.g. lazy projection and dictionary identity), then the consumer might not be aware of the load triggered by "lazy projection".